packages/blade/src/components/List/ListItem.tsx

```diff
@@ -0,0 +1,146 @@
+};
+
+const StyledListItem = styled(ListItemElement)<{
+  level?: number;
+  variant: NonNullable<ListProps['variant']>;
+  hasIcon: boolean;
+}>(({ level, theme, variant, hasIcon }) => ({
+  paddingLeft: level
+    ? getIn(
+        theme,
+        listItemPaddingLeft[
+          `${variant}${variant === 'unordered' && hasIcon ? 'WithIcon' : ''}` as NonNullable<
+            ListProps['variant'] | 'unorderedWithIcon'
+          >
+        ][level],
+      )
+    : 0,
+}));
+
+const ListItem = ({ children, icon: Icon, _itemNumber }: ListItemProps): React.ReactElement => {
+  const { level, size, icon: ListContextIcon, variant } = useListContext();
+  const { theme, platform } = useTheme();
+  const ItemIcon = Icon ?? ListContextIcon;
+
+  const childrenArray = React.Children.toArray(children);
+
+  const childItem = childrenArray.filter((child) =>
+    isValidAllowedChildren(child, 'List') ? null : child,
+  );
+  const childList = childrenArray.filter((child) =>
+    isValidAllowedChildren(child, 'List') ? child : null,
+  );
+  const hasIcon = Boolean(ItemIcon);
+
+  return (
+    <StyledListItem
+      level={level}
+      variant={variant}
+      hasIcon={hasIcon}
+      {...metaAttribute(MetaConstants.Component, MetaConstants.ListItem)}
+    >
+      <Box
+        display="flex"
+        flexDirection="row"
+        alignItems="center"
+        marginBottom={listItemPaddingBottom}
+      >
+        {variant === 'unordered' ? (
+          <Box
+            paddingRight={listItemBulletPaddingRight[variant]}
+            marginTop={
+              listItemBulletMarginTop[`${variant}${hasIcon ? 'WithIcon' : ''}`][platform][size]
+            }
+            display="flex"
+            alignSelf="flex-start"
+          >
+            {ItemIcon ? (
+              <ItemIcon size={size} color="surface.text.subdued.lowContrast" />
+            ) : (
+              <UnorderedItemIcon level={level} />
+            )}
+          </Box>
+        ) : (
+          <Box
+            width={listItemOrderedBulletBoxSize[variant][platform][size]}
+            height={listItemOrderedBulletBoxSize[variant][platform][size]}
+            marginRight={listItemBulletPaddingRight[variant]}
+            marginTop={listItemBulletMarginTop[variant][platform][size]}
+            display="flex"
+            flexShrink={0}
+            justifyContent="center"
+            alignSelf="flex-start"
+            alignItems="center"
+            borderRadius={variant === 'ordered-filled' ? 'max' : undefined}
+            backgroundColor={
+              variant === 'ordered-filled'
+                ? getIn(theme, 'colors.brand.gray.a100.lowContrast')
+                : undefined
+            }
+          >
+            {variant === 'ordered' ? (
+              <Text variant="body" type="subtle" size={size}>
+                {`${getOrderedListItemBullet({
+                  itemNumber: _itemNumber ?? 1,
+                  level: level ?? 1,
+                })}.`}
```

Comment:
Why is there a `.` in here but not in the else branch?

Reply:
Thats as per design. I've cleaned it up a bit now to make it more readable. `ordered` variant has a `.` but `ordered-filled` variant doesn't have a `.`